scipy/linalg/tests/test_lapack.py

```diff
@@ -632,158 +632,150 @@ def test_sgesdd_lwork_bug_workaround():
-        for dtype in REAL_DTYPES:
-            # Assert that a 0x0 matrix raises an error
-            A = np.zeros((0, 0), dtype=dtype)
-            sytrd, sytrd_lwork = \
-                get_lapack_funcs(('sytrd', 'sytrd_lwork'), (A,))
-            assert_raises(ValueError, sytrd, A)
-
-            # Tests for n = 1 currently fail with
-            # ```
-            # ValueError: failed to create intent(cache|hide)|optional array--
-            # must have defined dimensions but got (0,)
-            # ```
-            # This is a NumPy issue
-            # <https://github.com/numpy/numpy/issues/9617>.
-            # TODO Once the minimum NumPy version is past 1.14, test for n=1
-
-            # some upper triangular array
-            n = 3
-            A = np.zeros((n, n), dtype=dtype)
-            A[np.triu_indices_from(A)] = \
-                np.arange(1, n*(n+1)//2+1, dtype=dtype)
-
-            # query lwork
-            lwork, info = sytrd_lwork(n)
-            assert_equal(info, 0)
-
-            # check lower=1 behavior (shouldn't do much since the matrix is
-            # upper triangular)
-            data, d, e, tau, info = sytrd(A, lower=1, lwork=lwork)
-            assert_equal(info, 0)
-
-            assert_allclose(data, A, atol=5*np.finfo(dtype).eps, rtol=1.0)
-            assert_allclose(d, np.diag(A))
-            assert_allclose(e, 0.0)
-            assert_allclose(tau, 0.0)
-
-            # and now for the proper test (lower=0 is the default)
-            data, d, e, tau, info = sytrd(A, lwork=lwork)
-            assert_equal(info, 0)
-
-            # assert Q^T*A*Q = tridiag(e, d, e)
-
-            # build tridiagonal matrix
-            T = np.zeros_like(A, dtype=dtype)
-            k = np.arange(A.shape[0])
-            T[k, k] = d
-            k2 = np.arange(A.shape[0]-1)
-            T[k2+1, k2] = e
-            T[k2, k2+1] = e
-
-            # build Q
-            Q = np.eye(n, n, dtype=dtype)
-            for i in range(n-1):
-                v = np.zeros(n, dtype=dtype)
-                v[:i] = data[:i, i+1]
-                v[i] = 1.0
-                H = np.eye(n, n, dtype=dtype) - tau[i] * np.outer(v, v)
-                Q = np.dot(H, Q)
-
-            # Make matrix fully symmetric
-            i_lower = np.tril_indices(n, -1)
-            A[i_lower] = A.T[i_lower]
-
-            QTAQ = np.dot(Q.T, np.dot(A, Q))
-
-            # disable rtol here since some values in QTAQ and T are very close
-            # to 0.
-            assert_allclose(QTAQ, T, atol=5*np.finfo(dtype).eps, rtol=1.0)
+    @pytest.mark.parametrize('dtype', REAL_DTYPES)
+    def test_sytrd_with_zero_dim_array(self, dtype):
+        # Assert that a 0x0 matrix raises an error
+        A = np.zeros((0, 0), dtype=dtype)
+        sytrd = get_lapack_funcs('sytrd', (A,))
+        assert_raises(ValueError, sytrd, A)
+
+    @pytest.mark.parametrize('dtype', REAL_DTYPES)
+    @pytest.mark.parametrize('n', (1, 3))
+    def test_sytrd(self, dtype, n):
```

Comment:
I've verified that this function is unchanged except for the removal of the n=0 test and n=1 comment